lib/miq_environment.rb

```diff
@@ -8,10 +8,18 @@ def self.fully_qualified_domain_name
     Socket.gethostbyname(Socket.gethostname).first
   end
 
-  # Return the local IP v4 address of the local host, ignoring private addresses.
+  # Return the local IP v4 address of the local host
   #
   def self.local_ip_address
-    Socket.ip_address_list.detect { |addr| addr.ipv4? && !addr.ipv4_private? }&.ip_address
+    ipv4_addrs = Socket.ip_address_list.select(&:ipv4?)
+
+    # Prioritize "public" aka non-loopback non-private addresses first, then
+    # prefer private addresses, then take whatever we can get
+    local_addr   = ipv4_addrs.detect { |ip| !ip.ipv4_loopback? && !ip.ipv4_private? }
+    local_addr ||= ipv4_addrs.detect { |ip| !ip.ipv4_loopback? }
+    local_addr ||= ipv4_addrs.first
```

Comment:
For funsies, I think a sort would work too?

```ruby
local_addr = ipv4_address.sort_by { |ip| ip.ipv4_private? ? 0 : (ip.ipv4_loopback? ? 1 : 2) }.first
```

Only thing I'm not sure about is if there are multiple private ips, I'm not sure if this is deterministic

Reply:
Yeah good point about the return order from `Socket` not being guaranteed

Reply:
Alright now we're sorting by the ip_address and added a spec test with multiple private_ips returned by `Socket.ip_address_list` in different orders.